staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope.go

```diff
@@ -94,21 +95,27 @@ func (t *envelopeTransformer) TransformFromStorage(data []byte, context value.Co
 		value.RecordCacheMiss()
 		key, err := t.envelopeService.Decrypt(encKey)
 		if err != nil {
-			return nil, false, fmt.Errorf("error while decrypting key: %q", err)
+			return nil, false, err
 		}
 		transformer, err = t.addTransformer(encKey, key)
 		if err != nil {
-			return nil, false, err
+			return nil, false, status.Error(codes.Internal, "failed to add transformer")
 		}
 	}
-	return transformer.TransformFromStorage(encData, context)
+
+	out, stale, err := transformer.TransformFromStorage(encData, context)
+	if err != nil {
```

Comment:
What if an error came from a kms-plugin with some specific status?
Would not this make every error from a plugin to have a status of Internal?
I think we need to use status.FromError here.

Reply:
this transformer is not envelope transformer. what we get from kms-plugin is from `envelopeService`.

i'll remove these as suggested by liggit@